### ISSUE TITLE
Error when executing RustRunnable or RustDebuggable

### ISSUE THREAD

**Issue by Kannen** [simrat39/rust-tools.nvim#49]
I have the following error:
```
E5108: Error executing lua ...plugged/rust-tools.nvim/lua/rust-tools/runnables.lua:39:
  bad argument #1 to 'ipairs' (table expected, got nil)
```

I am still running on the linux distro where the lua jit compiler does not support goto statements, maybe another lua compat trouble?

**Comment by Kannen** [comment]
From what I see, the bug is fixed by adding
```
if (args.cargoExtraArgs ~= nil) then
...
end
```
around blocks that access to `cargoExtraArgs` field

**Comment by Kannen** [comment]
There is probably something else wrong, maybe when rust_analyzer is set up. I do not see inlayHints too...

**Comment by simrat39** [comment]
Are you setting up rust_analyzer yourself? if so then don't do that since rust-tools does that automatically

**Comment by Kannen** [comment]
I did it, I have removed it, but it does not change the problem.

Inside a cargo project when I run `lua print(vim.inspect(vim.lsp.buf_get_clients()))`
I can see an item named `commands = { RustDebuggables = { <function 3>},...}` so I
think rust-analyzer is configured by rust-tools.

The `.cache/nvim/lsp.log` is filled of error messages containing "rps" "rusta-analyzer" "stderr" ... ide::inlay_hints::...", the problem probably comes from rust-analyzer.
